### ISSUE TITLE
java.lang.IndexOutOfBoundsException:

### ISSUE THREAD

**Issue by iBotasky** [PhilJay/MPAndroidChart#2849]
I got a `IndexOutOfBoundsException` in `Time Chart`, here is the detail:
```
Process: com.xxmassdeveloper.mpchartexample, PID: 17005
                                                                                    java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
                                                                                        at java.util.ArrayList.get(ArrayList.java:411)
                                                                                        at com.github.mikephil.charting.data.DataSet.getEntryForIndex(DataSet.java:286)
                                                                                        at com.github.mikephil.charting.utils.Transformer.generateTransformedValuesLine(Transformer.java:184)
                                                                                        at com.github.mikephil.charting.renderer.LineChartRenderer.drawValues(LineChartRenderer.java:547)
                                                                                        at com.github.mikephil.charting.charts.BarLineChartBase.onDraw(BarLineChartBase.java:263)
                                                                                        at android.view.View.draw(View.java:17232)
                                                                                        at android.view.View.updateDisplayListIfDirty(View.java:16197)
                                                                                        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3758)
                                                                                        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3738)
                                                                                        at android.view.View.updateDisplayListIfDirty(View.java:16160)
                                                                                        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3758)
                                                                                        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3738)
                                                                                        at android.view.View.updateDisplayListIfDirty(View.java:16160)
                                                                                        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3758)
                                                                                        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3738)
                                                                                        at android.view.View.updateDisplayListIfDirty(View.java:16160)
                                                                                        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3758)
                                                                                        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3738)
                                                                                        at android.view.View.updateDisplayListIfDirty(View.java:16160)
                                                                                        at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:676)
                                                                                        at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:682)
                                                                                        at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:796)
                                                                                        at android.view.ViewRootImpl.draw(ViewRootImpl.java:2989)
                                                                                        at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2783)
                                                                                        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2374)
                                                                                        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1364)
                                                                                        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6757)
                                                                                        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:923)
                                                                                        at android.view.Choreographer.doCallbacks(Choreographer.java:735)
                                                                                        at android.view.Choreographer.doFrame(Choreographer.java:667)
                                                                                        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:909)
                                                                                        at android.os.Handler.handleCallback(Handler.java:755)
                                                                                        at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                                        at android.os.Looper.loop(Looper.java:156)
                                                                                        at android.app.ActivityThread.main(ActivityThread.java:6524)
                                                                                        at java.lang.reflect.Method.invoke(Native Method)
                                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:941)
                                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:831)
```

**Comment by MGaetan89** [comment]
This is a duplicate of #2450

**Comment by almic** [comment]
fixed now in d67ea481af1e8528c617dfd404d0c7827b0134b2
